### ISSUE TITLE
WealthSimple get stocks routes

### ISSUE THREAD

**Issue by GabrielBottari** [ahmedsakr/wstrade-api#58]
Wealthsimple trade app allows to get its available stocks by categories or top lists. It would be nice to have those to know which stock are available to trade.

I started investigating the API spec. Will update once I know more.

**Comment by ahmedsakr** [comment]
Great, keep me updated. It definitely would be a good addition to wstrade-api. I'm nearly wrapped up with v1.0.0, but it could be a minor version in v1.x.0.

**Comment by GabrielBottari** [comment]
I'm having a hard time reverse engineering the calls. I tried intercepting them with a vpn middleware, but the Wealthsimple Trade app seems to detect it and block it. How did you achieve this? My next attempt otherwise would be to decompile their apk.

**Comment by ahmedsakr** [comment]
Yep, i've already tried using motmproxy before and it's protected against MOTM attacks (good news as a customer :) ).

I have used [this repo](https://github.com/MarkGalloway/wealthsimple-trade/) to create the API wrapper on. On top of of that, I just fiddled with the parameters and guessed the interface for some endpoints based on patterns.

**Comment by GabrielBottari** [comment]
I have found a few interesting endpoints:

The /securities endpoint allows much more specification:
example: https://trade-service.wealthsimple.com/securities/top_market_movers?type=gainers&limit=50&offset=150
top_market_movers can be replaced with most_watched, crypto and gainers can be replaced with losers, most_watched and most_active (type is only available for top_market_movers)

There's also the /security-groups/{securitygroupid}/securities which allows to fetch securities by categories. To get categories just call /security-groups.
For example, we wanted to get all ETFs, we can do so with a query like: https://trade-service.wealthsimple.com/security-groups/security-group-c6153b067083/securities?limit=50&offset=50 and just change the offset to paginate over all results.

There's also more things, so let me know if you'd like me to forward it to you somehow (over slack maybe).

Would you want to add those to the library? Otherwise I could add them, but it might take a bit longer as I am not familiar with the codebase.

**Comment by ahmedsakr** [comment]
Great findings! I personally think all that you listed would be great extensions to wstrade-api in a future version.

For now, I think any work will need to be paused while i wrap up v1.0.0. Version 0 is not as friendly or modular to allow these extensions, and I wouldn’t want your efforts to go to waste. I plan on documentating v1.0.0 extensively with tests and contribution documents to on-board new contributors.

We can sync up on this soon! I plan on having v1.0.0 out the door by the end of the month.

**Comment by GabrielBottari** [comment]
Alright let me know :)

**Comment by ahmedsakr** [comment]
Hey @GabrielBottari ,

I took a bit longer but I pushed v1 in last night. I think work on this can start. I am thinking all of this will belong in the `data` module perfectly. We can have a discussion about this. Feel free to hit me up on Discord (AhmedSakr#5719)

**Comment by GabrielBottari** [comment]
Wealthsimple trade is now available on web for everyone, so might be interesting to see whats available from the networks tab

**Comment by ahmedsakr** [comment]
Wow, this is amazing. This will open up the doors for using a web-socket instead of REST/HTTPS calls.
